cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -640,8 +640,8 @@ private void executeAll(Iterator<WorkflowNode> iterator, ApplicationSpecificatio
           workflowContext.setState(new ProgramState(ProgramStatus.KILLED, rootCause.getMessage()));
           break;
         }
-        workflowContext.setState(new ProgramState(ProgramStatus.FAILED, rootCause.getMessage()));
-        throw Throwables.propagate(rootCause);
+        workflowContext.setState(new ProgramState(ProgramStatus.FAILED, t.getMessage()));
+        throw Throwables.propagate(t);
```

Comment:
If it doesn't change a bunch of things, would be better to throw the exception directly instead of wrapping it in a runtime exception.

Reply:
Fixed